lib/util/error_code.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Util
+  # helps get information regarding current error codes.
+  class ErrorCode
+    def initialize(path)
+      @path = path
+      validate
+    end
+
+    def next_available
+      err_num = -1
+      File.foreach(@path) do |line|
+        match = line.match(/code: \"CONJ(?<num>[0-9]+)E\"/)
+        if match
+          err_num = [err_num, match[:num].to_i].max
+        end
+      end
+
+      puts(error_code(err_num+1))
+    end
+
+    def validate
+      msg = format("The following path:%s was not found", @path)
+      raise FileNotFound.new(msg) unless File.file?(@path)
```

Comment:
Yeah, this is just telling you the idiomatic way to write this would be:

```suggestion
      raise FileNotFound, msg unless File.file?(@path)
```